metering-billing/lambdas/src/main/java/com/amazon/aws/partners/saasfactory/metering/onboarding/BillingIntegration.java

```diff
@@ -271,58 +272,94 @@ private void provisionTenantInStripe(String tenantId, String planId) throws Stri
         SubscriptionPlan planEnum;
         try {
             planEnum = SubscriptionPlan.valueOf(planId.toLowerCase());
-            if (planEnum == SubscriptionPlan.product_none)  {
-                LOGGER.info("Plan of {} for tenant {} will not be setup for billing.", planEnum.name(), tenantId);
-                return;
-            }
         } catch (IllegalArgumentException e) {
             LOGGER.error("provisionTenantInStripe: No PlanID: " + planId + " matched to SubscriptionPlanEnum");
             throw new RuntimeException("provisionTenantInStripe: Invalid PlanID: '" + planId + "' in the event detail");
         }
 
-        final String customerId = "tenant-" + tenantId.split("-")[0];  //get first segment
-        LOGGER.info("provisionTenantInStripe: Setting up Tenant: " + customerId + " as a customer and creating subscription");
+        final String customerName = "tenant-" + tenantId.split("-")[0];  //get first segment
+        LOGGER.info("provisionTenantInStripe: Setting up Tenant: {} as a customer and creating subscription", customerName);
 
-        try {
-            //Only attempt to setup tenant if customer does not already exist for this tenant
-            try {
-                Customer customer = Customer.retrieve(customerId);
-                LOGGER.info("Customer : " + customerId + " already exists, setup of tenant: " + tenantId + " skipped");
+        Customer customer = getStripeCustomer(customerName);
```

Comment:
What happens if Stripe is down or times out when responding? Do we fail the entire tenant onboarding stack? Should we retry, either immediately or lazily?

Reply:
Agree that we should have some sort of async retry mechanism in place for 3rd party integrations, but we are currently safe from blowing up the entire tenant onboarding/tenant update flow since we invoke these billing service api's async currently.

Reply:
I thought if the Lambda errors out the message from EventBridge will be retried.